src/inc/quic_platform_winkernel.h

```diff
@@ -469,58 +469,235 @@ _CxPlatEventWaitWithTimeout(
 #define CxPlatEventWaitWithTimeout(Event, TimeoutMs) \
     (STATUS_SUCCESS == _CxPlatEventWaitWithTimeout(&Event, TimeoutMs))
 
+//
+// TODO: Publish and document the following APIs
+//
+
+// ACCESS_MASK
+#define IO_COMPLETION_QUERY_STATE   0x0001
+#define IO_COMPLETION_MODIFY_STATE  0x0002  // winnt
+#define IO_COMPLETION_ALL_ACCESS (STANDARD_RIGHTS_REQUIRED|SYNCHRONIZE|0x3) // winnt
+
+typedef struct _FILE_IO_COMPLETION_INFORMATION {
+    PVOID               KeyContext;
+    PVOID               ApcContext;
+    IO_STATUS_BLOCK     IoStatusBlock;
+} FILE_IO_COMPLETION_INFORMATION, *PFILE_IO_COMPLETION_INFORMATION;
+
+// copied from zwapi.h
+NTSYSAPI
+NTSTATUS
+NTAPI
+ZwCreateIoCompletion (
+    OUT         PHANDLE IoCompletionHandle,
+    IN          ACCESS_MASK DesiredAccess,
+    IN OPTIONAL POBJECT_ATTRIBUTES ObjectAttributes,
+    IN OPTIONAL ULONG Count
+    );
+
+NTSYSAPI
+NTSTATUS
+NTAPI
+ZwSetIoCompletion (
+    IN HANDLE IoCompletionHandle,
+    IN PVOID KeyContext,
+    IN PVOID ApcContext,
+    IN NTSTATUS IoStatus,
+    IN ULONG_PTR IoStatusInformation
+    );
+
+NTSYSAPI
+NTSTATUS
+NTAPI
+ZwRemoveIoCompletionEx (
+    IN HANDLE IoCompletionHandle,
+    OUT PFILE_IO_COMPLETION_INFORMATION IoCompletionInformation,
+    IN ULONG Count,
+    OUT PULONG NumEntriesRemoved,
+    IN PLARGE_INTEGER Timeout OPTIONAL,
+    IN BOOLEAN Alertable
+    );
+
+typedef struct _IO_MINI_COMPLETION_PACKET_USER IO_MINI_COMPLETION_PACKET_USER, *PIO_MINI_COMPLETION_PACKET_USER;
+
+typedef
+VOID
+IO_MINI_PACKET_CALLBACK_ROUTINE(
+    _In_ PIO_MINI_COMPLETION_PACKET_USER MiniPacket,
+    _In_opt_ PVOID Context
+);
+
+typedef IO_MINI_PACKET_CALLBACK_ROUTINE *PIO_MINI_PACKET_CALLBACK_ROUTINE;
+
+struct _IO_MINI_COMPLETION_PACKET_USER {
+    LIST_ENTRY ListEntry;
+    ULONG PacketType;
+    PVOID KeyContext;
+    PVOID ApcContext;
+    NTSTATUS IoStatus;
+    ULONG_PTR IoStatusInformation;
+    PIO_MINI_PACKET_CALLBACK_ROUTINE MiniPacketCallback;
+    PVOID Context;
+    BOOLEAN Allocated;
+};
```

Comment:
I think you can delete this and leave it undefined in here, since you only deal in pointers and never write to it.